### ISSUE TITLE
use full qualified reth_codecs:: path + syntax when deriving reth_codecs::Compact

### ISSUE THREAD

**Issue by mattsse** [paradigmxyz/reth#11862]
### Describe the feature

currently we expect that Compact trait is in scope:

https://github.com/paradigmxyz/reth/blob/bc43613be35f316304f7ce4a2225855ac26b5923/crates/storage/codecs/derive/src/compact/generator.rs#L34-L36

hence we have a bunch of these:

https://github.com/paradigmxyz/reth/blob/bc43613be35f316304f7ce4a2225855ac26b5923/crates/primitives/src/receipt.rs#L13-L14

we can get rid of this by using the fully qualified path and syntax in the code we generate.

we also need a new attribute so that we can use this in the reth_codes crate itself, e.g.:

https://github.com/paradigmxyz/reth/blob/bc43613be35f316304f7ce4a2225855ac26b5923/crates/storage/codecs/src/alloy/transaction/legacy.rs#L6-L9

this should be similar to serde crate attribute

https://serde.rs/container-attrs.html#crate

so that we can use in the quote!

```
let reth_codecs = reth_codecs_path(); // TODO needs to check for `crate` attribute
quote!{
 impl #reth_codes::Compact ...

}
```

This task is done if we can remove all feature gated `use reth_codecs::Compact` imports

https://github.com/paradigmxyz/reth/pull/11863 has some wip and can be taken over

**Comment by aroralanuk** [comment]
Can I take this one?

**Comment by AlbionaHoti** [comment]
if you're not able to work on this @aroralanuk, let me know, would love to work on this

**Comment by aroralanuk** [comment]
> if you're not able to work on this @aroralanuk, let me know, would love to work on this

Hey @AlbionaHoti, give me 1 more day. Otherwise, I'll unassign myself and ping you here.

**Comment by aroralanuk** [comment]
@AlbionaHoti Hey, I have a hectic week ahead, so I won't be able to close this one. I'm unassigned myself if you wanna give it a go. Feel free to look at my wip PR.

**Comment by mattsse** [comment]
cool, ty!
taking this over

**Comment by DanielEmmanuel1** [comment]
Can I try solving this issue?

Background:
I am a Full-Stack Blockchain developer with ample expirience writing Rust and Solidity. I have built efficient and scalable applications and worked with libraries and frameworks, and I have a solid understanding of Rust's features and idioms.

My Proposed Solution ToTackle The Issue:
1. Fully Qualified Paths: I'll will modify the code generation logic to use the fully qualified path for reth_codecs::Compact instead of relying on imports. This will reduce the clutter in the code.

2. New Attribute Introduction: I'll will then design and implement a new attribute that can be used within the reth_codecs crate. This should function similarly to attributes used in the serde crate.

3. Code Refactor: Then I will review and refactor existing code to replace all feature-gated use reth_codecs::Compact imports with the new fully qualified syntax.

4. Testing: I will ensure that all existing tests pass and add new tests to verify that the new attribute and fully qualified paths work as expected.

5. Documentation Update: Update the documentation to reflect these changes, including examples of how to use the new attribute.

**Comment by mattsse** [comment]
hey @AlbionaHoti do you still want to work on this?

**Comment by DanielEmmanuel1** [comment]
Can i be assigned this

**Comment by mattsse** [comment]
assigned @DanielEmmanuel1

**Comment by DanielEmmanuel1** [comment]
I'll get write on it @mattsse

**Comment by mattsse** [comment]
closed via #12279
